src/displayapp/screens/HeartRate.cpp

```diff
@@ -82,10 +82,14 @@ void HeartRate::Refresh() {
     case Controllers::HeartRateController::States::NoTouch:
     case Controllers::HeartRateController::States::NotEnoughData:
       // case Controllers::HeartRateController::States::Stopped:
-      lv_label_set_text_static(label_hr, "000");
+      lv_label_set_text_static(label_hr, "---");
       break;
     default:
-      lv_label_set_text_fmt(label_hr, "%03d", heartRateController.HeartRate());
+      if (heartRateController.HeartRate() == 0) {
+        lv_label_set_text_static(label_hr, "---");
```

Comment:
is this last if-else needed? doesn't the state go back to `NotEnoughData`? Or does it go back down to 0 heartrate? I'm curious

Reply:
It does look like it goes to `NotEnoughData`, at least, if I let it get a value then remove it from my skin, it displays the not enough data text.

Reply:
I'm quite sure it's not needed.

Reply:
it is in place so that when you press the stop button is displays "---" instead of "000", as this does not fall into the `NotEnoughData`

Reply:
Ah, thanks for the clarification!